Night CI 2023-04-20: Failed

action url: https://github.com/spidernet-io/spiderpool/actions/runs/4758501201

```tex
E0420 20:32:32.690884       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 378 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x293a300?, 0x3f0e0a0?})
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0xf3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0007606c0?})
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0xb0
panic({0x293a300, 0x3f0e0a0})
	/usr/local/go/src/runtime/panic.go:890 +0x263
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).handleIPPool.func1()
	/src/pkg/ippoolmanager/ippool_informer.go:270 +0x217
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).handleIPPool(0xc0007da300, {0x2fe1fc8, 0xc000130018}, 0xc0013d5a40)
	/src/pkg/ippoolmanager/ippool_informer.go:289 +0x1f4
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).processNextWorkItem.func1({0x284f620?, 0xc0014d66e0})
	/src/pkg/ippoolmanager/ippool_informer.go:225 +0x525
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).processNextWorkItem(0xc0007da300)
	/src/pkg/ippoolmanager/ippool_informer.go:258 +0x79
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).runWorker(...)
	/src/pkg/ippoolmanager/ippool_informer.go:188
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0001185d0)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:157 +0x49
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x2fc62a0, 0xc000608e70}, 0x1, 0xc000732ba0)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:158 +0xcf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xdf1b54?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x10e
k8s.io/apimachinery/pkg/util/wait.Until(0x420700?, 0x7ff077c21f00?, 0x0?)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:92 +0x49
created by github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).Run
	/src/pkg/ippoolmanager/ippool_informer.go:177 +0x26a
{"level":"DEBUG","ts":"2023-04-20T20:32:32.800Z","logger":"Subnet-Informer","caller":"subnetmanager/subnet_informer.go:241","msg":"Enqueue Subnet","IPPoolName":"v6pool-3231-905646462","SubnetName":"v6-ss-3231-178855849","Operation":"SYNC"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x265f577]

goroutine 378 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0007606c0?})
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0x145
panic({0x293a300, 0x3f0e0a0})
	/usr/local/go/src/runtime/panic.go:890 +0x263
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).handleIPPool.func1()
	/src/pkg/ippoolmanager/ippool_informer.go:270 +0x217
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).handleIPPool(0xc0007da300, {0x2fe1fc8, 0xc000130018}, 0xc0013d5a40)
	/src/pkg/ippoolmanager/ippool_informer.go:289 +0x1f4
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).processNextWorkItem.func1({0x284f620?, 0xc0014d66e0})
	/src/pkg/ippoolmanager/ippool_informer.go:225 +0x525
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).processNextWorkItem(0xc0007da300)
	/src/pkg/ippoolmanager/ippool_informer.go:258 +0x79
github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).runWorker(...)
	/src/pkg/ippoolmanager/ippool_informer.go:188
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0001185d0)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:157 +0x49
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x2fc62a0, 0xc000608e70}, 0x1, 0xc000732ba0)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:158 +0xcf
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xdf1b54?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x10e
k8s.io/apimachinery/pkg/util/wait.Until(0x420700?, 0x7ff077c21f00?, 0x0?)
	/src/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:92 +0x49
created by github.com/spidernet-io/spiderpool/pkg/ippoolmanager.(*IPPoolController).Run
	/src/pkg/ippoolmanager/ippool_informer.go:177 +0x26a
```

--- a/pkg/ippoolmanager/ippool_informer.go
+++ b/pkg/ippoolmanager/ippool_informer.go
@@ -264,14 +264,6 @@ func (ic *IPPoolController) processNextWorkItem() bool {
 }
 
 func (ic *IPPoolController) handleIPPool(ctx context.Context, pool *spiderpoolv2beta1.SpiderIPPool) (err error) {
-	defer func() {
-		if err == nil {
-			attr := attribute.String(constant.KindSpiderIPPool, pool.Name)
-			metric.IPPoolTotalIPCounts.Add(ctx, *pool.Status.TotalIPCount, attr)
-			metric.IPPoolAvailableIPCounts.Add(ctx, (*pool.Status.TotalIPCount)-(*pool.Status.AllocatedIPCount), attr)
-		}
-	}()
-
 	// checkout the Auto-created IPPools whether need to scale or clean up legacies
 	if ic.EnableSpiderSubnet && IsAutoCreatedIPPool(pool) {
 		err := ic.cleanAutoIPPoolLegacy(ctx, pool)
@@ -285,6 +277,10 @@ func (ic *IPPoolController) handleIPPool(ctx context.Context, pool *spiderpoolv2
 	if nil != err {
 		return err
 	}
+
+	attr := attribute.String(constant.KindSpiderIPPool, pool.Name)
+	metric.IPPoolTotalIPCounts.Add(ctx, *pool.Status.TotalIPCount, attr)
+	metric.IPPoolAvailableIPCounts.Add(ctx, (*pool.Status.TotalIPCount)-(*pool.Status.AllocatedIPCount), attr)
 
 	return nil
 }

--- a/pkg/subnetmanager/subnet_informer.go
+++ b/pkg/subnetmanager/subnet_informer.go
@@ -354,13 +354,6 @@ func (sc *SubnetController) syncHandler(ctx context.Context, subnetName string) 
 	}
 
 	subnetCopy := subnet.DeepCopy()
-	defer func() {
-		if err == nil {
-			attr := attribute.String(constant.KindSpiderSubnet, subnetName)
-			metric.SubnetTotalIPCounts.Add(ctx, *subnetCopy.Status.TotalIPCount, attr)
-			metric.SubnetAvailableIPCounts.Add(ctx, (*subnetCopy.Status.TotalIPCount)-(*subnetCopy.Status.AllocatedIPCount), attr)
-		}
-	}()
 
 	if err := sc.syncMetadata(ctx, subnetCopy); err != nil {
 		return fmt.Errorf("failed to sync metadata of Subnet: %v", err)
@@ -379,6 +372,10 @@ func (sc *SubnetController) syncHandler(ctx context.Context, subnetName string) 
 			return fmt.Errorf("failed to remove finalizer: %v", err)
 		}
 	}
+
+	attr := attribute.String(constant.KindSpiderSubnet, subnetName)
+	metric.SubnetTotalIPCounts.Add(ctx, *subnetCopy.Status.TotalIPCount, attr)
+	metric.SubnetAvailableIPCounts.Add(ctx, (*subnetCopy.Status.TotalIPCount)-(*subnetCopy.Status.AllocatedIPCount), attr)
 
 	return nil
 }

